### ISSUE TITLE
Add voting for random/captains for a full queue

### ISSUE THREAD

**Issue by mattwells19** [mattwells19/UNCC-Six-Mans.js#20]
**Note: This feature is considered out of scope for MVP. Therefore, items required for MVP should be completed before completing this item.**

## Description
Right now only one player in the queue needs to select "captains" or "random" for the selection to be made. This can be unfavorable if most people want to pick captains, but the one player who wants random clicks the random button first.

To alleviate this issue, we want to implement functionality similar to "broken queue" which requires a majority vote before making a selection.

As people vote, the random/captains buttons should update to reflect the number of votes for each option.

## Technical Details
Because there is only one queue, the registered votes do not need to be stored in the database. Broken queue votes are stored in the db since there can be multiple active matches that can be voted on.

My suggestion would be to add a local, private variable to the `QueueRepository` that tracks the votes for each selection. Something like a `Map` where the key is the id of the player and the value is the selection (random/captains). Then, when one of the selections reaches a majority vote, 4 for a 6 person queue, that action is then executed.

In other words, once 4 people vote for captains, captains are chosen and team picking begins.

## Acceptance Criteria
 - When the queue is full and a player NOT in the queue makes a selection, nothing happens.
 - When the queue is full and a player who is in the queue makes a selection, their vote is recorded and the button they pushed increments the vote count by one.
   - For ex: Button text before: `Captains (0)`, button text after being pressed: `Captains (1)`.
 - Once a selection reaches the majority vote, the action of the selection is performed.
 - If a player in the queue leaves during the selection process, all votes are discarded.
